.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,42 @@
+name: Unit test on push and PR
+
+
+on: [push, pull_request, workflow_dispatch]
+
+
+jobs:
+  unit-tests:
+    name: Run Unit Tests
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [windows-latest, ubuntu-latest, macos-latest]
```

Comment:
Should non-Windows OS be listed here?

Reply:
yeah we're using this code in Colab's and potentially cloud work in the future, so the unit tests ensuring functionality should be run on, in addition to Windows, at least also Linux. Adding macOS isn't directly necessitated, but imo we may as well just to be well-rounded. (Also I often develop on macOS, so I personally would appreciate it.)